go/base/management/commands/go_assign_tagpool.py

```diff
@@ -8,69 +8,100 @@ class Command(BaseGoAccountCommand):
-                    raise CommandError('Please provide %s:' % (opt.dest,))
-
-        self.handle_validated(*args, **options)
-
-    def handle_validated(self, *args, **options):
         tagpool = unicode(options['tagpool'])
-        max_keys = int(options['max_keys']) or None
 
         account = self.user_api.get_user_account()
-        existing_tagpools = []
+        existing_perms = []
         for permissions in account.tagpools.load_all_bunches():
-            existing_tagpools.extend([
+            existing_perms.extend([
                 p for p in permissions if p.tagpool == tagpool])
 
-        if existing_tagpools:
-            if options['update']:
-                if len(existing_tagpools) == 1:
-                    self.update_permission(existing_tagpools[0], max_keys)
-                else:
-                    raise CommandError('%s permissions specified for '
-                        'the same tagpool. Please fix manually' % (
-                            len(existing_tagpools),))
-            else:
-                raise CommandError('Permission already exists, use '
-                    '--update to update the value of max-keys')
+        if len(existing_perms) > 1:
+            raise CommandError(
+                '%s permissions specified for the same tagpool. Please fix'
+                ' manually' % (len(existing_perms),))
+
+        if options['update']:
+            self.process_update_permission(
+                account, tagpool, existing_perms, options)
+        elif options['remove']:
+            self.process_remove_permission(
+                account, tagpool, existing_perms, options)
         else:
-            self.create_permission(account, tagpool, max_keys)
+            self.process_create_permission(
+                account, tagpool, existing_perms, options)
+
+    def process_update_permission(self, account, tagpool, perms, options):
+        if not perms:
+            raise CommandError(
+                'Could not update permission. Tagpool permission not found.')
+        if not options['max_keys']:
+            raise CommandError(
+                'Updating a permission requires setting the max_keys options.')
+        max_keys = int(options['max_keys']) or None
+        self.update_permission(perms[0], max_keys)
 
     def update_permission(self, tagpool_permission, max_keys):
         tagpool_permission.max_keys = max_keys
         tagpool_permission.save()
 
+    def process_remove_permission(self, account, tagpool, perms, options):
+        if not perms:
+            raise CommandError(
+                'Could not remove permission. Tagpool permission not found.')
+        if options['max_keys'] is not None:
+            raise CommandError(
+                'The max-keys option is may not be set when when deleting a'
+                ' permission.')
+        self.remove_permission(perms[0])
+
+    def remove_permission(self, tagpool_permission):
+        tagpool_permission.delete()
```

Comment:
Just to follow the pattern of function-for-processing-options and function-for-doing-work pattern that the other commands use.